apps/prelaunch/models.py

```diff
@@ -66,7 +63,16 @@ def get_or_create_by_email(
         If `enable_early_access` is `True`, will also enable for early access.
         """
         try:
-            prelaunch_signup = cls.objects.get(email=email, site=site)
+            prelaunch_signups = cls.objects.filter(email=email, site=site)
+            prelaunch_signups_with_early_access = prelaunch_signups.filter(
+                early_access=True
+            )
+            prelaunch_signup = (
+                prelaunch_signups_with_early_access.first()
+                if prelaunch_signups_with_early_access.count() > 0
+                else prelaunch_signups.first()
+            )
```

Comment:
```suggestion
            prelaunch_signup = (
                prelaunch_signups_with_early_access.first()
                or prelaunch_signups.first()
            )
```

- `.first()` will return `None` if it does not exist, so it is always safe to call
- `.exists()` is a better pattern and preferred over `.count() > 0`, because the latter actually performs a `SELECT COUNT…`

Reply:
You are right. Updated.